### ISSUE TITLE
qwen3-30b-a3b可以用transformers-version 5.2.0做sft训练吗？训练后用sglang-version 0.5.9推理好像有问题

### ISSUE THREAD

**Issue by yuliangCarmelo** [QwenLM/Qwen2.5#1827]
### Description

qwen3-30b-a3b可以用transformers-version 5.2.0做sft训练吗？训练后用sglang-version 0.5.9推理好像有问题

### Reproduction

sft

### Logs

```shell

```

### Environment Information

训练环境
absl-py                           2.4.0
accelerate                        1.12.0
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.3
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-doc                     0.0.4
annotated-types                   0.7.0
anthropic                         0.71.0
antlr4-python3-runtime            4.9.3
anyio                             4.12.1
apache-tvm-ffi                    0.1.8.post2
apex                              0.1
astor                             0.8.1
asttokens                         3.0.1
attrdict                          2.0.1
attrs                             25.4.0
audioread                         3.1.0
auto_gptq                         0.7.1
av                                16.1.0
binpacking                        2.0.1
bitsandbytes                      0.49.1
blake3                            1.0.8
brotli                            1.2.0
cachetools                        7.0.1
cbor2                             5.8.0
certifi                           2026.1.4
cffi                              2.0.0
cfgv                              3.5.0
charset-normalizer                3.4.4
click                             8.3.1
cloudpickle                       3.1.2
colorama                          0.4.6
colorlog                          6.10.1
compressed-tensors                0.12.2
contourpy                         1.3.3
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      46.0.5
cuda-bindings                     13.1.1
cuda-pathfinder                   1.3.4
cuda-python                       13.1.1
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.6.0
decorator                         5.2.1
decord                            0.6.0
deepspeed                         0.17.6
depyf                             0.20.0
diffusers                         0.36.0
dill                              0.3.8
diskcache                         5.6.3
distlib                           0.4.0
distro                            1.9.0
dnspython                         2.8.0
docstring_parser                  0.17.0
dotenv                            0.9.9
editdistance                      0.8.1
einops                            0.8.1
email-validator                   2.3.0
et_xmlfile                        2.0.0
evalscope                         1.4.2
evaluate                          0.4.6
executing                         2.2.1
fastapi                           0.129.0
fastapi-cli                       0.0.21
fastapi-cloud-cli                 0.12.0
fastar                            0.8.0
fastrlock                         0.8.3
ffmpy                             1.0.0
filelock                          3.20.2
fire                              0.7.1
flash-attn                        2.8.3
flash-linear-attention            0.4.2
flashinfer-python                 0.5.3
fonttools                         4.61.1
frozenlist                        1.8.0
fsspec                            2025.3.0
funasr                            1.3.1
func-timeout                      4.3.5
fuzzywuzzy                        0.18.0
gekko                             1.3.2
genson                            1.3.0
gguf                              0.17.1
gitdb                             4.0.12
GitPython                         3.1.46
google-auth                       2.48.0
google-genai                      1.63.0
gradio                            5.32.1
gradio_client                     1.10.2
groovy                            0.1.2
grpcio                            1.78.0
h11                               0.16.0
h5py                              3.15.1
hf-xet                            1.2.0
hjson                             3.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
httpx-sse                         0.4.3
huggingface_hub                   1.5.0
human-eval                        1.0.3
hydra-core                        1.3.2
icecream                          2.1.8
identify                          2.6.16
idna                              3.11
ijson                             3.4.0.post0
ImageIO                           2.37.2
immutabledict                     4.3.0
importlib_metadata                8.7.1
interegular                       0.3.3
ipdb                              0.13.13
ipython                           9.10.0
ipython_pygments_lexers           1.1.1
jaconv                            0.5.0
jamo                              0.4.1
jedi                              0.19.2
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.13.0
jmespath                          0.10.0
joblib                            1.5.3
json_repair                       0.57.1
json5                             0.13.0
jsonlines                         4.0.0
jsonpath-ng                       1.7.0
jsonschema                        4.26.0
jsonschema-specifications         2025.9.1
kaldiio                           2.18.1
kiwisolver                        1.4.9
langdetect                        1.0.9
lark                              1.2.2
latex2sympy2_extended             1.11.0
lazy_loader                       0.4
Levenshtein                       0.27.3
librosa                           0.11.0
liger_kernel                      0.7.0
lightning-utilities               0.15.2
llguidance                        1.3.0
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
lmdeploy                          0.10.1
loguru                            0.7.3
lxml                              6.0.2
Markdown                          3.10.2
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
math-verify                       0.9.0
matplotlib                        3.10.8
matplotlib-inline                 0.2.1
mcp                               1.26.0
mdurl                             0.1.2
megatron-core                     0.15.4
mistral_common                    1.9.1
ml_dtypes                         0.5.4
mmengine-lite                     0.10.7
model-hosting-container-standards 0.1.13
modelscope                        1.33.0
more-itertools                    10.8.0
mpmath                            1.3.0
ms-agent                          1.5.2
ms-opencompass                    0.1.6
ms-vlmeval                        0.0.19
msgpack                           1.1.2
msgspec                           0.20.0
multidict                         6.7.0
multiprocess                      0.70.16
networkx                          3.6.1
ninja                             1.13.0
nltk                              3.9.2
nodeenv                           1.10.0
numba                             0.61.2
numpy                             1.26.4
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cudnn-frontend             1.18.0
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-cutlass-dsl                4.4.0
nvidia-cutlass-dsl-libs-base      4.4.0
nvidia-ml-py                      13.590.48
nvidia-nccl-cu12                  2.27.5
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvshmem-cu12               3.3.20
nvidia-nvtx-cu12                  12.8.90
nvitop                            1.6.2
omegaconf                         2.3.0
onnx                              1.20.1
onnx-ir                           0.1.16
onnxscript                        0.6.2
openai                            2.21.0
openai-harmony                    0.0.8
openai-whisper                    20250625
OpenCC                            1.2.0
opencv-python                     4.11.0.86
opencv-python-headless            4.13.0.92
openpyxl                          3.1.5
optimum                           2.1.0
orjson                            3.11.7
oss2                              2.19.1
outlines                          1.2.11
outlines_core                     0.2.11
overrides                         7.7.0
packaging                         25.0
pandas                            2.3.3
parso                             0.8.6
partial-json-parser               0.2.1.1.post7
peft                              0.18.1
pexpect                           4.9.0
pillow                            11.3.0
pip                               23.3.2
platformdirs                      4.8.0
ply                               3.11
pooch                             1.9.0
portalocker                       3.2.0
pre_commit                        4.5.1
prettytable                       3.17.0
prometheus_client                 0.24.1
prometheus-fastapi-instrumentator 7.1.0
prompt_toolkit                    3.0.52
propcache                         0.4.1
protobuf                          6.33.5
psutil                            7.2.2
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
py-spy                            0.4.1
pyarrow                           22.0.0
pyasn1                            0.6.2
pyasn1_modules                    0.4.2
pybase64                          1.4.3
pybind11                          3.0.1
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pydantic                          2.11.10
pydantic_core                     2.33.2
pydantic-extra-types              2.11.0
pydantic-settings                 2.12.0
pydub                             0.25.1
pyecharts                         2.1.0
Pygments                          2.19.2
PyJWT                             2.11.0
pylatexenc                        2.10
pynndescent                       0.6.0
pynvml                            13.0.1
pyparsing                         3.3.2
pypinyin                          0.55.0
python-dateutil                   2.9.0.post0
python-dotenv                     1.2.1
python-json-logger                4.0.0
python-Levenshtein                0.27.3
python-multipart                  0.0.22
pytorch-wpe                       0.0.1
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-omni-utils                   0.0.9
qwen-vl-utils                     0.0.14
rank-bm25                         0.2.2
RapidFuzz                         3.14.3
ray                               2.53.0
referencing                       0.37.0
regex                             2025.11.3
requests                          2.32.5
rich                              13.9.4
rich-toolkit                      0.19.4
rignore                           0.7.6
rouge                             1.0.1
rouge-chinese                     1.0.3
rouge-score                       0.1.2
rpds-py                           0.30.0
rsa                               4.9.1
ruff                              0.15.1
sacrebleu                         2.6.0
safehttpx                         0.1.7
safetensors                       0.7.0
scikit-learn                      1.8.0
scipy                             1.16.3
seaborn                           0.13.2
semantic-version                  2.10.0
sentence-transformers             5.2.2
sentencepiece                     0.2.1
sentry-sdk                        2.52.0
setproctitle                      1.3.7
setuptools                        65.5.1
shellingham                       1.5.4
shortuuid                         1.0.13
simplejson                        3.20.2
six                               1.17.0
smmap                             5.0.2
sniffio                           1.3.1
sortedcontainers                  2.4.0
soundfile                         0.13.1
soxr                              1.0.0
sse-starlette                     3.2.0
stack-data                        0.6.3
starlette                         0.52.1
sty                               1.0.6
supervisor                        4.3.0
swanlab                           0.7.8
sympy                             1.14.0
tabulate                          0.9.0
tenacity                          9.1.4
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
tensorboardX                      2.6.4
termcolor                         3.3.0
threadpoolctl                     3.6.0
tiktoken                          0.12.0
timeout-decorator                 0.5.0
timm                              1.0.24
tokenizers                        0.22.2
tomlkit                           0.13.3
torch                             2.9.0
torch-complex                     0.4.4
torchaudio                        2.9.0
torchmetrics                      1.8.2
torchvision                       0.24.0
tqdm                              4.67.1
traitlets                         5.14.3
transformer_engine                2.11.0
transformer_engine_cu12           2.11.0
transformer-engine-torch          2.11.0
transformers                      5.2.0         /root/paddlejob/workspace/env_run/output/transformers-5.2.0
transformers-stream-generator     0.0.5
triton                            3.5.0
trl                               0.24.0
typer                             0.23.1
typer-slim                        0.23.1
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.3
umap-learn                        0.5.11
urllib3                           2.6.3
uv                                0.10.9
uvicorn                           0.40.0
uvloop                            0.22.1
validators                        0.35.0
virtualenv                        20.36.1
vllm                              0.13.0
wandb                             0.25.0
watchfiles                        1.1.1
wcwidth                           0.6.0
websockets                        15.0.1
Werkzeug                          3.1.5
wheel                             0.45.1
word2number                       1.1
wrapt                             2.1.1
xgrammar                          0.1.27
xlsxwriter                        3.2.9
xxhash                            3.6.0
yapf                              0.43.0
yarl                              1.22.0
zhconv                            1.4.3
zipp                              3.23.0
zstandard                         0.25.0

### Known Issue

- [ ] The issue hasn't been already addressed in Documentation, Issues, and Discussions.
